components/service/glean/src/main/java/mozilla/components/service/glean/BooleanMetricType.kt

```diff
@@ -58,8 +62,16 @@ data class BooleanMetricType(
      *                 value in [sendInPings]
      * @return true if metric value exists, otherwise false
      */
+    @Suppress("MagicNumber")
     @VisibleForTesting(otherwise = VisibleForTesting.NONE)
     fun testHasValue(pingName: String = getStorageNames().first()): Boolean {
+        if (ioTask != null) {
+            var loopCnt = 50
+            while (ioTask!!.isActive && loopCnt > 0) {
+                --loopCnt
+            }
+        }
```

Comment:
If `loopCnt` is `0` here, maybe log a message to the console that we timed out waiting for metric recording to finish.  That will help sort out failures for some other reason on the line below from failures due to timing out.  (And if we started seeing a lot of those messages next to seemingly random failures, we'd know that we'd need to revisit this solution.)

Reply:
Using `withTimeout` made this a little more difficult to accomplis.   It should throw a TimeoutCancellationException which I suppose we could catch and log, but since this is only used from a test then maybe throwing is better than logging?